packages/grafana-runtime/src/accesscontrol/authorize.ts

```diff
@@ -0,0 +1,104 @@
+import { locationService, type BackendSrv } from '../services';
+
+type PermissionMap = Record<string, boolean>;
+
+export interface Authorizer {
+  getSignedInUserPermissions(reload: boolean): Promise<PermissionMap>;
+  hasPermission(principal: AuthorizationPrincipal, action: string): boolean;
+  hasAnyPermission(principal: AuthorizationPrincipal, actions: string[]): boolean;
+  hasAllPermissions(principal: AuthorizationPrincipal, actions: string[]): boolean;
+}
+
+interface AuthorizationPrincipal {
+  permissions?: PermissionMap;
+}
+
+function hasPermission(principal: AuthorizationPrincipal, action: string): boolean {
+  return !!principal.permissions?.[action];
+}
+
+/**
+ * DefaultAuthorizer is one that makes authorization decisions based on passed information.
+ * It would be up to the user of DefaultAuthorizer to decide how to store and provide authorization
+ * information to checks.
+ */
+export class DefaultAuthorizer {
```

Comment:
We can enforce `DefaultAuthorizer` to implement interface. It's might be good for static type checks to make sure it implements it correctly.
```suggestion
export class DefaultAuthorizer implements Authorizer {
```

Reply:
Nice, thanks - made my own commit (https://github.com/grafana/grafana/pull/91164/commits/92b45e10365949baf0d137a6a349b9928e6a84bc) to add it to both authorizers.